shard/shard.go

```diff
@@ -80,6 +80,18 @@ type Shard interface {
 	// ID returns the ID of the shard
 	ID() uint32
 
+	// CutoverNanos returns when shard traffic is cut over.
+	CutoverNanos() int64
```

Comment:
Feels weird to have the same interface used by `m3db` and `m3aggregator` if these fields are only used by one of them. Should they be separated?

Reply:
The reason why they live in the `Shard` interface is that the placement algorithm already works with the shard interface, so if we separate them out into different types, we need to either duplicate existing algorithm code to work with a different type, or have type casts in various places in the algorithm to do different things based on the shard type, which is also messy.

Adding these two methods to the shard interface makes the shards slightly more generic and is also non-intrusive to support the shard-level cutover / cutoff we need without significantly rewriting the algorithms. Even For `m3db`, if it needs staged shard cutover or cutoff at some point for whatever reason, it's readily supported.